pkgs/by-name/gi/git-butler/package.nix

```diff
@@ -0,0 +1,71 @@
+{ lib, stdenv, fetchurl, fetchzip, appimageTools, undmg }:
+let
+  pname = "git-butler";
+  version = "0.11.7";
+  build_version = "889";
+
+  src = {
+    x86_64-darwin = fetchurl {
+      url =
+        "https://releases.gitbutler.com/releases/release/${version}-${build_version}/macos/x86_64/GitButler_${version}_x64.dmg";
```

Comment:
I don't understand what you mean by an internet archive. Can you give me an example please? I've used this structure instead of a list a url because the `.appimage` is compressed inside a tarball. I can't use directly `fetchurl` without a lot of boilerplate which is already included in `fetchzip`.

Reply:
Something like [this](https://github.com/NixOS/nixpkgs/blob/a2ab4d5ae9cbda096b021273eb9faefbf0065e5e/pkgs/by-name/av/avr-sim/package.nix#L24)

[Archiving the page](https://web.archive.org/save) on the internet archive will provide a fallback if the gitbutler site should go down at some point.

Reply:
oh ok, I haven't seen this kind of pattern before. Updating right away

Reply:
@msanft I see one problem with this method. The IA url isn't dynamic. The `data` part in the url cannot be determined at when building like `version` and `build_version` whose are dynamic. One line to change and all urls are up-to-date regardless of the system arch. I'm not quite sure it is an appropriate way when the base url can be dynamic across all systems.

Reply:
The IA URL needs to be re-created anyways for each version bump (as you'd need to save the URL with the new version). The same applies for the hash too? 

I don't object keeping it like it currently is though if you have a strong opinion on this.

Reply:
I've updated the PR. I've made a mix between the two methods. Lets see what's happening next :+1: